extensions/query-store/src/reports/queryStoreDashboard.ts

```diff
@@ -8,41 +8,53 @@ import * as azdata from 'azdata';
 import * as constants from '../common/constants';
 import { TopResourceConsumingQueries } from './topResourceConsumingQueries';
 import { OverallResourceConsumption } from './overallResourceConsumption';
+import { Deferred } from '../common/promise';
 
 export class QueryStoreDashboard {
+	private dashboard?: azdata.window.ModelViewDashboard;
+	private initDashboardComplete: Deferred = new Deferred();
+
 	constructor(private dbName: string, private extensionContext: vscode.ExtensionContext) { }
 
 	/**
 	 * Creates and opens the report
 	 */
 	public async open(): Promise<void> {
 		// TODO: update title based on selected tab to have the current selected report in editor tab title
-		const dashboard = azdata.window.createModelViewDashboard(constants.queryStoreDashboardTitle(this.dbName));
-		dashboard.registerTabs(async (view: azdata.ModelView) => {
+		this.dashboard = azdata.window.createModelViewDashboard(constants.queryStoreDashboardTitle(this.dbName));
+		this.dashboard.registerTabs(async (view: azdata.ModelView) => {
 			const topResourceConsumingQueriesReport = new TopResourceConsumingQueries(this.extensionContext, this.dbName);
 			const overallResourceConsumptionReport = new OverallResourceConsumption(this.extensionContext, this.dbName);
 
 			await Promise.all([topResourceConsumingQueriesReport.createReport(view), overallResourceConsumptionReport.createReport(view)]);
 
 			const topResourceConsumingQueriesTab: azdata.DashboardTab = {
-				id: 'TopResourceConsumingQueriesTab',
+				id: constants.topResourceConsumingQueriesTabId,
 				content: topResourceConsumingQueriesReport.ReportContent!,
 				title: constants.topResourceConsumingQueries
 			};
 
 			const overallResourceConsumptionTab: azdata.DashboardTab = {
-				id: 'OverallResourceConsumptionTab',
+				id: constants.overallResourceConsumptionTabId,
 				content: overallResourceConsumptionReport.ReportContent!,
 				title: constants.overallResourceConsumption
 			};
 
+			this.initDashboardComplete?.resolve();
+
 			return [
 				overallResourceConsumptionTab,
 				topResourceConsumingQueriesTab
 			];
 		});
 
-		await dashboard.open();
+		await this.dashboard.open();
+		await this.initDashboardComplete;
```

Comment:
had to add this so that the tabs are finished being registered by the time `selectTab()` gets called from extension.ts